mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -45,37 +48,76 @@ pub async fn disable_coin(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, St
     };
-    let swaps = try_s!(active_swaps_using_coin(&ctx, &ticker));
-    if !swaps.is_empty() {
-        let err = json!({
-            "error": format!("There're active swaps using {}", ticker),
-            "swaps": swaps,
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
+    let coins_ctx = CoinsContext::from_ctx(&ctx).map_err(|err| ERRL!("{}", err))?;
+
+    // If a platform coin is to be disabled, we get all the enabled tokens for this platform coin first.
+    let mut coins_to_disable = coins_ctx.get_tokens_to_disable(&ticker).await;
+    // We then add the platform coin to the end of the list of the coins to be disabled.
+    coins_to_disable.push(ticker.clone());
+    drop_mutability!(coins_to_disable);
+
+    // Get all matching orders and active swaps.
+    let mut active_swaps = vec![];
+    let mut still_matching_orders = vec![];
+    for ticker in &coins_to_disable {
+        let swaps = try_s!(active_swaps_using_coin(&ctx, ticker));
+        let still_matching = try_s!(get_matching_orders(&ctx, ticker).await);
+        active_swaps.extend(swaps);
+        still_matching_orders.extend(still_matching);
     }
-    let (cancelled, still_matching) = try_s!(cancel_orders_by(&ctx, CancelBy::Coin { ticker: ticker.clone() }).await);
-    if !still_matching.is_empty() {
+    drop_mutability!(active_swaps);
+    drop_mutability!(still_matching_orders);
+
+    // If there're matching orders or active swaps we return an error.
+    if !active_swaps.is_empty() || !still_matching_orders.is_empty() {
         let err = json!({
-            "error": format!("There're currently matching orders using {}", ticker),
-            "orders": {
-                "matching": still_matching,
-                "cancelled": cancelled,
-            }
+            "error": format!("There're currently matching orders or active swaps for some tokens"),
+            "active_swaps": active_swaps,
+            "still_matching": still_matching_orders
         });
         return Response::builder()
-            .status(500)
+            .status(INTERNAL_SERVER_ERROR_CODE)
             .body(json::to_vec(&err).unwrap())
             .map_err(|e| ERRL!("{}", e));
     }
 
-    try_s!(disable_coin_impl(&ctx, &ticker).await);
+    // Proceed with diabling the coin/tokens.
+    let mut cancelled_orders = vec![];
+    let mut disabled_tokens = vec![];
+    for ticker in &coins_to_disable {
+        log!("disabling {ticker} coin");
+        let coin = match lp_coinfind(&ctx, ticker).await {
+            Ok(Some(t)) => t,
+            Ok(None) => return ERR!("No such coin: {}", ticker),
+            Err(err) => return ERR!("!lp_coinfind({}): ", err),
+        };
+        let (cancelled, _) = try_s!(
+            cancel_orders_by(&ctx, CancelBy::Coin {
+                ticker: ticker.to_string()
+            })
+            .await
+        );
```

Comment:
I believe we also need to return a response in this case
```rust
json!({
                "error": err,
                "cancelled_orders": cancelled_orders,
                "disabled_tokens": disabled_tokens,
})
```

Please consider adding a function similar to:
```rust
async fn cancel_and_remove_coin(coin: &str) -> Result<(Vec<Uuid>, Vec<Uuid>), String> { todo!() }

// An example how it can be used:
let mut cancelled_orders = vec![];
let mut disabled_tokens = vec![];
for ticker in coins_to_disable.iter() {
  match cancel_and_remove_coin(ticker).await {
    Ok((cancelled, _)) => cancelled_orders.extend(cancelled),
    Err(e) => {
        let err = json!({
                "error": err,
                "cancelled_orders": cancelled_orders,
                "disabled_tokens": disabled_tokens,
        });
        return Response::builder()
                .status(INTERNAL_SERVER_ERROR_CODE)
                .body(json::to_vec(&err).unwrap())
                .map_err(|e| ERRL!("{}", e));
    }
  }
}
```

Reply:
done